destination/destination_test.go

```diff
@@ -204,52 +169,38 @@ func TestS3Parquet(t *testing.T) {
 		t.Fatalf("failed to initialize destination: %v", err)
 	}
 
-	for _, record := range generateRecords(50) {
-		err = destination.WriteAsync(ctx, record, getAckFunc())
+	// generate 50 records and write them in 2 batches
+	records := generateRecords(50)
+	count, err := destination.Write(ctx, records[:25])
+	is.NoErr(err)
+	is.Equal(count, 25)
 
-		if err != nil {
-			t.Fatalf("Write returned an error: %v", err)
-		}
-	}
+	count, err = destination.Write(ctx, records[25:])
+	is.NoErr(err)
+	is.Equal(count, 25)
 
 	writer, ok := destination.Writer.(*writer.S3)
-
-	if !ok {
-		t.Fatalf("Destination writer expected to be writer.S3, but is actually %+v", writer)
-	}
-
-	err = destination.Flush(ctx)
-	if err != nil {
-		t.Fatalf("failed to Flush: %v", err)
-	}
+	is.True(ok) // Destination writer expected to be writer.S3
 
 	err = destination.Teardown(ctx)
-	if err != nil {
-		t.Fatalf("failed to Teardown: %v", err)
-	}
+	is.NoErr(err)
 
 	// check if only two files are written
-	if len(writer.FilesWritten) != 2 {
-		t.Fatalf("Expected writer to have written 2 files, got %d", len(writer.FilesWritten))
-	}
+	is.Equal(len(writer.FilesWritten), 2) // Expected writer to have written 2 files
 
 	validator := &filevalidator.S3{
-		AccessKeyID:     awsAccessKeyID,
-		SecretAccessKey: awsSecretAccessKey,
-		SessionToken:    awsSessionToken,
```

Comment:
@maha-hajja IIRC the session token isn't really needed, right?

Reply:
yeah we don't need `awsSessionToken`.. we might think about adding it later as an optional config.. because some users might want to test the s3 connector using temporary tokens that would expire after some time.